pool/app/session/entity.ml

```diff
@@ -436,16 +436,52 @@ module Calendar = struct
   type location =
     { id : Pool_location.Id.t
     ; name : Pool_location.Name.t
+    ; url : string
     }
   [@@deriving eq, show, yojson]
 
+  type links =
```

Comment:
type links =
    { experiment : string option
    ; session : string option
    }

Would't that suffice? Where session is either the normal session link or location session link, depending on the role?

Reply:
Selecting the correct link is done in a second step. First, it just checks for all sessions you've access to and then when preparing the calendar UI it checks whether you'll see the experiment/session or location/session link.